azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventHubClient.java

```diff
@@ -134,57 +134,86 @@ public EventHubClient apply(MessagingFactory factory) {
     }
 
     /**
-     * Creates an Empty Collection of {@link EventData}.
-     * The same partitionKey must be used while sending these events using {@link EventHubClient#send(Iterable)}.
+     * BatchBuilder is used to create {@link EventDataBatch}es. When creating an {@link EventDataBatch} from an
+     * {@link EventHubClient}, the user is able to set the partitionKey and maxMessageSize. The default partitionKey is null,
+     * and when a null partitionKey is used, {@link EventData} within the {@link EventDataBatch} will land in every
+     * EventHub partition via the Round-robin algorithm.
      *
-     * @param partitionKey PartitionKey used while actually sending the Events.
-     * @return the empty {@link EventDataBatch}, after negotiating maximum message size with EventHubs service
-     * @throws EventHubException if the Microsoft Azure Event Hubs service encountered problems during the operation.
+     * <pre>
+     *     {@code
+     *     // Create EventDataBatch with default properties
+     *     EventDataBatch edb1 = ehClient.new BatchBuilder().createBatch();
+     *
+     *     // Create EventDataBatch with user-defined partitionKey and maxMessageSize
+     *     EventDataBatch edb2 = ehClient.new BatchBuilder().with( options -> {
+     *         options.partitionKey = "foo";
+     *         options.maxMessageSize = 256;
+     *     }).createBatch();
+     *
+     *     // Create EventDataBatch with user-defined partitionKey
+     *     EventDataBatch edb3 = ehClient.new BatchBuilder()
+     *          .with( options -> options.partitionKey = "foo")
+     *          .createBatch();
```

Comment:
We should come up with an API which is also possible in .Net. This way we are defeating our purpose of providing similar API across clients.